lambda_functions/user_service/get_user_ranking.py

```diff
@@ -0,0 +1,89 @@
+import logging
+import json
+import traceback
+
+from core_layer.connection_handler import get_db_session
+from core_layer.handler import user_handler
+from core_layer import helper
+
+
+def get_user_ranking(event, context, is_test=False, session=None):
+    """
+    returns a dictionary with
+    list top_users:           10 top users
+    list top_users_by_level:  10 top users on the user's level 
+    list top_users_by_period: 10 top users in a period p (1 week)
+    """
+    n = 10
+    p = 1
+    
+    descending = True
+    attr = 'experience_points'
+    
+    logger = logging.getLogger()
+    logger.setLevel(logging.INFO)
+    helper.log_method_initiated("Get User Ranking", event, logger)
+
+    if session == None:
+        session = get_db_session(False, None)
+
+    user_ranking_dict = {}
+
+    top_users_list = []
+    top_users_by_level_list = []
+    top_users_by_period_list = []
+
+    helper.log_method_initiated("Get Top n Users", event, logger)     
+    try:
+        top_users = user_handler.get_top_users(n, attr, descending, is_test, session)
+        for user in top_users:            
+            top_users_list.append(user.to_dict())
+        user_ranking_dict['top_users'] = top_users_list
+    except Exception:
```

Comment:
Putting all three handler calls into one try/except block could save you a few lines of code and make the file a bit easier to understand.

Reply:
Yes! If it's not necessary to have a specific try/except block for each handler call it's better arranged in a single block.
Just kept an extra try/except around the `cognito_id_from_event()` call.